fdm-calculator/src/doses/calculate-dose.test.ts

```
@@ -0,0 +1,103 @@
+            p_dose_n: 2500,
+            p_dose_p2o5: 1250,
+            p_dose_k2o: 625,
+        }
+        expect(calculateDose({ applications, fertilizers })).toEqual(
+            expectedDose,
+        )
+    })
+
+    it("should handle missing fertilizer data gracefully", () => {
+        const applications = [{ p_id: "fertilizer1", p_app_amount: 100 }]
+        const fertilizers: any[] = [] // No matching fertilizer
```

Biome (1.9.4)

[error] Unexpected any. Specify a different type.

any disables many type checking rules. Its use should be avoided.

(lint/suspicious/noExplicitAny)